fil_fungible_token/src/token/mod.rs

```diff
@@ -524,23 +521,50 @@ mod test {
     const BOB: &Address = &Address::new_id(4);
     const CAROL: &Address = &Address::new_id(5);
 
-    fn new_token() -> Token<MemoryBlockstore, FakeMessenger> {
-        Token::new(MemoryBlockstore::default(), FakeMessenger::new(TOKEN_ACTOR.id().unwrap(), 6))
-            .unwrap()
-            .0
+    fn new_token(
+        bs: MemoryBlockstore,
+        state: &mut TokenState,
+    ) -> Token<MemoryBlockstore, FakeMessenger> {
+        Token::new(bs, FakeMessenger::new(TOKEN_ACTOR.id().unwrap(), 6), state).unwrap()
     }
 
     fn assert_last_msg_eq(messenger: &FakeMessenger, expected: TokenReceivedParams) {
         let last_called = messenger.last_hook.borrow().clone().unwrap();
         assert_eq!(last_called, expected);
     }
 
+    #[test]
+    fn it_wraps_a_previously_loaded_state_tree() {
+        struct ActorState {
+            token_state: TokenState,
+        }
+
+        // simulate the token state being a node in a larger state tree
+        let bs = MemoryBlockstore::default();
+        let mut actor_state = ActorState { token_state: TokenState::new(&bs).unwrap() };
+
+        // wrap the token state, moving it into a TokenHandle
+        let mut token = Token::new(
+            &bs,
+            FakeMessenger::new(TOKEN_ACTOR.id().unwrap(), 6),
+            &mut actor_state.token_state,
+        )
+        .unwrap();
+
+        token.mint(TOKEN_ACTOR, TREASURY, &TokenAmount::from(1), &Default::default()).unwrap();
+
+        let state = token.state();
+        // gets a read-only state
```

Comment:
Can we access `actor_state` or is that now disallowed due to the mutable reference? (Fine if not, but please demonstrate). If we can access it, can we show the change is visible there too, ie. there's only one copy?

Reply:
`actor_state.token_state` can still be accessed (see line 560 now) but doing so invalidates the `token` created on line 547. 

not quite what I expected/wanted on first consideration but makes sense that it works like this.